### ISSUE TITLE
Test failure: `com.linecorp.armeria.server.DomainSocketServerTest.executionError`

### ISSUE THREAD

**Issue by ikhoon** [line/armeria#4959]
```
java.io.IOException: Failed to delete temp directory /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/junit9515291060972060513. The following paths could not be deleted (see suppressed exceptions for details): test.sock
	at org.junit.jupiter.engine.extension.TempDirectory$CloseablePath.createIOExceptionWithAttachedFailures(TempDirectory.java:351)
	at org.junit.jupiter.engine.extension.TempDirectory$CloseablePath.close(TempDirectory.java:252)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.execution.ExtensionValuesStore.lambda$closeAllStoredCloseableValues$3(ExtensionValuesStore.java:68)
```

**Comment by ikhoon** [comment]
Similar but produced multiple times on different builds.
https://github.com/line/armeria/actions/runs/5748084789?pr=4644

**Comment by trustin** [comment]
That doesn't sound right. `DomainSocketServerTest` doesn't use `TempDirectory` but `TemporaryFolderExtension`.

**Comment by trustin** [comment]
Fixed already by @jrhee17 at ea3b063f3ab870881e6e16b46ccb583902d97a6e :sweat_smile:

**Comment by ikhoon** [comment]
This problem seemed not fixed with `TemporaryFolderExtension`. https://github.com/line/armeria/actions/runs/5816889007/job/15770694528?pr=5108#step:7:1585 
After checking Netty code, I found out a race condition where Netty tries to delete a domain socket file when `ServerDomainSocketChannel` is closed.
https://github.com/netty/netty/blob/9eb029d630b269e907aaa4a19bc86753d70e399c/transport-classes-epoll/src/main/java/io/netty/channel/epoll/EpollServerDomainSocketChannel.java#L79-L81
